lib/identity_cache/query_api.rb

```diff
@@ -228,7 +228,7 @@ def preload_id_embedded_associations(records)
           scope = child_model.all
           scope = scope.instance_exec(nil, &reflection.scope) if reflection.scope
 
-          pairs = scope.where(reflection.foreign_key => records.map(&:id)).pluck(reflection.foreign_key, reflection.active_record_primary_key)
+          pairs = scope.where(reflection.foreign_key => records.map(&:id)).pluck(reflection.foreign_key, child_model.primary_key)
```

Comment:
@rafaelfranca this looks okay to me, but can you think of how this can backfire?